code/modules/library/book.dm

```diff
@@ -0,0 +1,355 @@
+		return
+	pages += content
+
+/obj/item/book/proc/add_page(content)
+	var/text = content
+	if(!text)
+		text = " "
+	if(length(pages) >= MAX_PAGES)
+		return FALSE
+	pages += text
+	current_page = length(pages) //open to newely added page so player can edit it
+	return TRUE
+
+/obj/item/book/proc/remove_page(page_number)
+	if(length(pages) < page_number) //can't remove the page if it doesn't exist
+		return FALSE
+	if(pages -= pages[page_number])
+		page_number = length(pages) //if page_number is somehow at a value it shouldn't be we fix it here aswell
+		return TRUE //we want to make sure whatever is calling this proc knows the operation was succesful
+	return FALSE
+
+/obj/item/book/proc/carve_book(mob/user, obj/item/I)
+	if(carved)
+		to_chat(user, "<span class='warning>[title] has already been carved out!</span>")
+		return
+	if(!I.sharp)
+		to_chat(user, "<span class='warning>You can't carve [title] using that!</span>")
+		return
+	to_chat(user, "<span class='notice'>You begin to carve out [title].</span>")
+	if(I.use_tool(src, user, 30, volume = I.tool_volume))
+		user.visible_message("<span class='warning'>[user] appears to carve out the pages inside of [title]!</span>",\
+				"<span class='danger'>You begin to carve out [title].!</span>")
+		carved = TRUE
+		return TRUE
+
+/obj/item/book/proc/store_item(obj/item/I, mob/user)
+	if(store)
+		to_chat(user, "<span class='notice'>There is already something in [name]!</span>")
+		return
+
+	//does it exist, if so is it an abstract item?
+	if(!I || !istype(I) || (I.flags & ABSTRACT))
+		return
+	if(I.flags & NODROP)
+		to_chat(user, "<span class='notice'>[I] stays stuck to your hand when you try and hide it in the book!.</span>")
+	//Checking to make sure the item we're storing isn't larger than/equal to size of the book, prevents recursive storing aswell
+	if(I.w_class >= w_class)
+		to_chat(user, "<span class='notice'>[I] is to large to fit in [name].</span>")
+		return
+
+	user.drop_item()
+	I.forceMove(src)
+	store = I
+	to_chat(user, "<span class='notice'>You hide [I] in [name].</span>")
+
+
+/obj/item/book/proc/remove_stored_item(mob/user, thrown = FALSE)
+	if(!store)
+		if(!thrown) //we don't wanna display this message if the object falls out because it was thrown
+			to_chat(user, "<span class='notice'>You search [name] but there is nothing in it!</span>")
+		return FALSE
+	if(!thrown)
+		to_chat(user, "<span class='notice'>You carefully remove [store] from [name]!</span>")
+	store.forceMove(get_turf(store.loc))
+	store = null
+	return TRUE
+
+ //* Book Spawners n'stuff *//
+/obj/item/book/random
+	icon_state = "random_book"
+	var/amount = 1
+
+/obj/item/book/random/Initialize()
+	..()
+	var/list/books = GLOB.library_catalog.getRandomBooks(amount)
+	for(var/b in books)
+		var/datum/cachedbook/book = b
+		var/obj/item/book/B = new(loc) //this really should be a constructor that accepts cachedbook datums at some point tbh
```

Comment:
Would this PR be a good time to make that constructor?

Reply:
yes, this is one of a few little notes I left while I was coding at 2AM b/c I was too tired to implement it

Reply:
ok I've put it in, although I need to also implement it with the book binder and library computer as well soon